api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/aup/AcceptableUsagePolicyProperties.java

```diff
@@ -64,6 +64,32 @@
     @RequiredProperty
     private String aupAttributeName = "aupAccepted";
 
+    /**
+     * Scope options for the default aup repository can store flag indicating acceptance.
+     * Scope refers to duration that acceptance is kept.
+     * Current options are global on the particular server (not replicated across CAS servers)
+     * and once per authentication via credentials (not authentication events via TGT).
+     */
+    public enum DefaultAupScope {
+        /**
+         * Store in global map for life of server.
+         */
+        GLOBAL_SERVER,
+
+        /**
+         * Store in flow scope and flow only reaches aup when user authenticates, not when they login via TGT.
+         * Using flow scope allows the AUP flow to finish and then AUP flow will not be invoked again until
+         * another authentication occurs.
+         */
+        AUTHENTICATION
+    };
+
+    /**
+     * Scope of map where the aup selection is stored.
+     */
+    @RequiredProperty
```

Comment:
You might want to take out an annotation. It indicates whether something is required * by the user * to configure. In our case, it's not since you're supplying a default value/behavior.

Reply:
btw, the setting should be also documented.